nptdms/writer.py

```diff
@@ -30,29 +59,63 @@ def __init__(self, file, mode='w', version=4712):
             It's important that if you are appending segments to an
             existing TDMS file, this matches the existing file version (this can be queried with the
             :py:attr:`~nptdms.TdmsFile.tdms_version` property).
+        :param index_file: Whether or not to write a index file besides the data file. Index files
+            can be used to accelerate reading speeds for faster channel extraction and data positions inside
+            the data files. Only valid if submitted file variable is a path.
         """
         valid_versions = (4712, 4713)
         if version not in valid_versions:
             raise ValueError("version must be one of %s" % ",".join("%d" % v for v in valid_versions))
         self._file = None
+        self._index_file = None
         self._file_path = None
+        self._index_file_path = None
         self._file_mode = mode
         self._tdms_version = version
 
         if hasattr(file, "read"):
             # Is a file
             self._file = file
+            if isinstance(index_file, BytesIO):
```

Comment:
Rather than checking if this is specifically a `BytesIO` instance, we should allow any readable object.
```suggestion
            if hasattr(index_file, "read"):
```